scripts/generate_sudo_list.py

```diff
@@ -0,0 +1,74 @@
+"""Helper script to generate sudo list doc."""
+import argparse
+import filecmp
+import sys
+import tempfile
+from pathlib import Path
+
+import yaml
+
+
+def collect_sudo_commands(path=None):
+    """Collect all ansible cmds that require sudo."""
+    if not path:
+        path = Path("quipucords/scanner/network/runner")
+    list_cmds = []
+    for playbook in path.glob("roles/*/tasks/main.yml"):
+        for task in yaml.safe_load(playbook.open()):
+            if task.get("become", ""):
+                list_cmds.append(task.get("raw", ""))
+    list_cmds = [
+        cmd.strip() for cmd in list_cmds if isinstance(cmd, str) and cmd.strip()
+    ]
+    list_cmds = sorted(set(list_cmds))
+
+    return list_cmds
+
+
+def generate_sudo_cmds_document():
+    """Generate a document listing commands that require sudo."""
+    list_cmds = collect_sudo_commands()
+    document = ""
+    document += "\n".join(cmd for cmd in list_cmds if cmd.strip())
+    return document
+
+
+def create_document_file(file_path):
+    """Create a txt file containing a list of commands that require sudo."""
+    document = generate_sudo_cmds_document()
+    file = Path(file_path)
+    file.write_text(document)
+
+
+def compare_sudo_cmds_lists(file_path):
+    """Compare the most updated sudo list with an existing one."""
+    generated_doc = generate_sudo_cmds_document()
+
+    with tempfile.NamedTemporaryFile(delete=False) as temp_file:
+        temp_file.write(generated_doc.encode())
+        try:
+            if filecmp.cmp(temp_file.name, file_path):
+                sys.stdout.write("The sudo command list document is up to date.\n")
```

Comment:
why write to stdout instead of stderr?

Reply:
Since `sys.stdout `is used for normal program output, while `sys.stderr `is used for error messages, I thought would be nice to differentiate the messages. Was that a bad choice ?

Reply:
unix way is to output any type of message to stderr
you only output to stdout if this the "output" of the program. this is a log for a script without an output

we could also just print nothing if nothing is wrong (I tend to prefer that XD, but that's just me)

Reply:
if it were to write anything to stdout, I'd say write the contents it would write to a file

not saying to do that, tho. just something that would make sense on stdout.

Reply:
I will change to print nothing if everything is ok.